src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/helper/ContactDetailsValidatorTest.java

```diff
@@ -0,0 +1,89 @@
+package uk.gov.hmcts.reform.finrem.caseorchestration.helper;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.Address;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.FinremCaseData;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.YesOrNo;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.wrapper.ContactDetailsWrapper;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.helper.ContactDetailsValidator.APPLICANT_POSTCODE_ERROR;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.helper.ContactDetailsValidator.APPLICANT_SOLICITOR_POSTCODE_ERROR;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.helper.ContactDetailsValidator.RESPONDENT_POSTCODE_ERROR;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.helper.ContactDetailsValidator.RESPONDENT_SOLICITOR_POSTCODE_ERROR;
+
+
+class ContactDetailsValidatorTest {
+
+    @ParameterizedTest
+    @MethodSource("provideInvalidCaseData")
+    void shouldReturnErrorWhenPostcodeIsMissing(FinremCaseData caseData, String expectedError) {
+        List<String> errors = ContactDetailsValidator.validateCaseDataAddresses(caseData);
+        assertEquals(1, errors.size(), "Expected only one error.");
```

Comment:
If you went for this
`assertThat(errors).containsOnly(expectedError);`
Then there'd be no need for all those messages